lib/internal/Magento/Framework/View/Layout/Argument/Interpreter/DataObject.php

```diff
@@ -44,13 +58,17 @@ public function evaluate(array $data)
         if (!isset($data['value'])) {
             throw new \InvalidArgumentException('Object class name is missing.');
         }
+
+        $shared = isset($data['shared']) ? $this->booleanUtils->toBoolean($data['shared']) : true;
```

Comment:
In order to maintain backward compatibility please set this attribute to be `false` by default